assets/tests/hooks/useApiCall.test.ts

```diff
@@ -0,0 +1,184 @@
+import { describe, expect, jest, test } from "@jest/globals"
+import { act, renderHook, waitFor } from "@testing-library/react"
+import { useApiCall } from "../../src/hooks/useApiCall"
+
+const renderUseApiCall = (initialProps: Parameters<typeof useApiCall>[0]) =>
+  renderHook(useApiCall, {
+    initialProps,
+  })
+
+const PromiseWithResolvers = <T>() => {
+  let resolve: (value: T) => void = () => {}
+  let reject: (reason?: any) => void = () => {}
+  const promise = new Promise((res, rej) => {
+    resolve = res
+    reject = rej
+  })
+
+  if (resolve === undefined || reject === undefined) {
```

Comment:
I'm trying to understand this. Will this non-deterministically detect a race condition (or something) where `resolve` or `reject` get set to `undefined`?

Reply:
This was just to make typescript happy since I didn't have access to [Promise.withResolvers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers)

Reply:
I don't actually know why I don't have access to that, especially in a _test_ environment, but I wasn't getting anywhere and didn't want to spend more time on it. I assume it has something to do with the package.json or something.

Reply:
Oh uh, I don't think I need this anymore since I tried to give `resolve` and `reject` default values to try and get rid of this `if` statement. Not sure what would be better.

Reply:
Oooh I see - it's to get Typescript to recognize those values as not-`undefined`!!

This is wounding my brain, but is there any way to know that the function passed in to `Promise`'s constructor will get called before the if statement happens?

I guess I dunno whether the if statement is better, or whether the default values are better... if there's some surprise and `resolve`/`reject` don't get initialized, then the test will be easier to debug with the if statement (cause it'll fail there), although I would perhaps update the message to be something more verbose than `Error("Bad")` 😂

Reply:
😭 
```
const PromiseWithResolvers = <T>() => {
  let resolve: undefined | ((v: T) => void) = undefined
  let reject: undefined | ((v: any) => void) = undefined
  const promise = new Promise<T>((res, rej) => {
    resolve = res
    reject = rej
  })

  if (resolve === undefined || reject === undefined) {
    throw Error("Bad")
  }

  return { promise, resolve, reject }
}
```
```
    tests/hooks/useApiCall.test.ts:130:7 - error TS2349: This expression is not callable.
      Type 'never' has no call signatures.

    130       resolve("first result")
```

Reply:
Gosh I wish we had `Promise.withResolvers` 😞